Testing logging seems broken in pytest 8

pytest 8 seems to break some of our tests, and perhaps our code as well. The order of the tests suddenly matter, and it is related to logging:
```
pip install pytest==8.2.1
```
This works fine:
```
python -m pytest -x --pdb \
    scopesim/tests/tests_effects/test_Quantization.py::TestApplyTo::test_logs_warning_for_float_dtype \
    scopesim/tests/test_logging.py::test_log_to_file 
```
This fails:
```
python -m pytest -x --pdb \
    scopesim/tests/test_logging.py::test_log_to_file \
    scopesim/tests/tests_effects/test_Quantization.py::TestApplyTo::test_logs_warning_for_float_dtype
```
with
```
==== test session starts =====
platform linux -- Python 3.11.5, pytest-8.2.1, pluggy-1.5.0
rootdir: .../ScopeSim
configfile: pyproject.toml
plugins: cov-5.0.0, anyio-4.3.0
collected 2 items                                                                                                                          

scopesim/tests/test_logging.py .                                                                                                     [ 50%]
scopesim/tests/tests_effects/test_Quantization.py F
>>>> captured stdout >>>>
astar.scopesim.effects.electronic - WARNING: Cannot access cmds for Quantization effect.
astar.scopesim.effects.electronic - WARNING: Setting quantized data to dtype <class 'float'>, which is not an integer subtype.
>>>> traceback >>>>

self = <scopesim.tests.tests_effects.test_Quantization.TestApplyTo object at 0x7f014d714a10>
det = <scopesim.detector.detector.Detector object at 0x7f014d70f2d0>, caplog = <_pytest.logging.LogCaptureFixture object at 0x7f014d70a5d0>

    def test_logs_warning_for_float_dtype(self, det, caplog):
        assert det.data.sum() > det.data.size * 1.2
        quant = Quantization(dtype=float)
        det = quant.apply_to(det)
        assert det.data.sum() == det.data.size
        # Test sub-dtype because exact dtype for "float" may depend on OS
        assert np.issubdtype(det.data.dtype, np.floating)
>       assert "Setting quantized data to dtype" in caplog.text
E       AssertionError: assert 'Setting quantized data to dtype' in ''
E        +  where '' = <_pytest.logging.LogCaptureFixture object at 0x7f014d70a5d0>.text

scopesim/tests/tests_effects/test_Quantization.py:56: AssertionError
>>>> entering PDB >>>>

>>>> PDB post_mortem (IO-capturing turned off) >>>
> .../ScopeSim/scopesim/tests/tests_effects/test_Quantization.py(56)test_logs_warning_for_float_dtype()
-> assert "Setting quantized data to dtype" in caplog.text
```

--- a/scopesim/tests/conftest.py
+++ b/scopesim/tests/conftest.py
@@ -16,8 +16,14 @@ MOCK_DIR = Path(__file__).parent / "mocks"
 sim.rc.__currsys__["!SIM.file.error_on_missing_file"] = True
 
 
-@pytest.fixture(scope="package", autouse=True)
+@pytest.fixture(scope="function", autouse=True)
 def configure_logging():
+    """This disables the handlers so the logs reach pytests' caplog.
+
+    This fixture should be on the function level, because some functions
+    might call `update_logging()` and therefore undo this fixture for
+    subsequent tests. E.g. test_log_to_file() does this.
+    """
     base_logger = logging.getLogger("astar")
     handlers = base_logger.handlers
     # Disable handlers

